sklearn/compose/_column_transformer.py

```diff
@@ -803,6 +847,10 @@ def _hstack(self, Xs):
             return sparse.hstack(converted_Xs).tocsr()
         else:
             Xs = [f.toarray() if sparse.issparse(f) else f for f in Xs]
+            config = get_output_config("transform", self)
```

Comment:
`fit_transform` will also called this function and we decide to ouput only depending on `transform`. It is a bit inconsistent. I don't know if we should change the code or just revise the SLEP to have one entry in the config for `fit_transform` and `transform`. Basically, we will never have anything for `fit` since it returns only `self`.

Reply:
I think this is expected behavior. If someone sets the output of `transform`, then I think it's okay to assume they want `fit_transform` to also have the same output.

For me, having to set both keys is a worst user experience.